packages/next/bin/next.ts

```diff
@@ -107,7 +106,10 @@ if (process.env.NODE_ENV) {
 ;(process.env as any).NODE_ENV = process.env.NODE_ENV || defaultEnv
 ;(process.env as any).NEXT_RUNTIME = 'nodejs'
 
-if (shouldUseReactRoot) {
+// In node.js runtime, react has to be required after NODE_ENV is set,
+// so that the correct dev/prod bundle could be loaded into require.cache.
+const React = require('react')
+if (parseInt(React.version) >= 18) {
```

Comment:
Can we do `const { shouldUseReactRoot } = require('../server/utils')` here to reuse the logic?